src/Library/demos/Launcher/main.js

```diff
@@ -86,6 +88,7 @@ uri_launch.connect("clicked", () => {
   launchesUri().catch(logError);
 });
 
-uri_details.connect("clicked", () => {
-  console.log(`File: ${uri_launcher.get_uri()}`);
+uri_details.connect("changed", () => {
+  text = uri_details.get_text();
+  uri_launcher.set_uri(text);
```

Comment:
```suggestion
  uri_launcher.set_uri(text);
  
  try {
    uri_launch.sensitive = GLib.Uri.is_valid(text, GLib.UriFlags.NONE);
  } catch {
    uri_launch.sensitive = false;
  }
```

Let's tie the sensitivity of the button to the URI.